lib/status/status.go

```diff
@@ -212,10 +212,15 @@ type ClusterOperationProgress struct {
 	Created time.Time `json:"created"`
 }
 
+// GetSystemStatus returns the status of the system
+func (r Agent) GetSystemStatus() pb.SystemStatus_Type {
+	return pb.SystemStatus_Type(r.SystemStatus)
+}
+
 // Agent specifies the status of the system and individual nodes
 type Agent struct {
 	// SystemStatus defines the health status of the whole cluster
-	SystemStatus pb.SystemStatus_Type `json:"system_status"`
+	SystemStatus SystemStatus `json:"system_status"`
```

Comment:
I moved this to a variable of an alias type to be able to format it for display (see `String` and `GoString` overrides below), so that system status comes out as `degraded` instead of `2`.